src/EditorFeatures/Core/Navigation/AbstractDefinitionLocationService.cs

```diff
@@ -119,24 +130,136 @@ internal abstract class AbstractDefinitionLocationService : IDefinitionLocationS
         // instead navigate to the actual interface member.
         //
         // In the future we can expand this with other mappings if appropriate.
-        var interfaceImpls = symbol.ExplicitOrImplicitInterfaceImplementations();
-        if (interfaceImpls.Length == 0)
-            return null;
+        return await TryGetExplicitInterfaceLocationAsync().ConfigureAwait(false) ??
+               await TryGetInterceptedLocationAsync().ConfigureAwait(false);
+
+        async ValueTask<INavigableLocation?> TryGetExplicitInterfaceLocationAsync()
+        {
+            var interfaceImpls = symbol.ExplicitOrImplicitInterfaceImplementations();
+            if (interfaceImpls.Length == 0)
+                return null;
 
-        var title = string.Format(EditorFeaturesResources._0_implemented_members,
-            FindUsagesHelpers.GetDisplayName(symbol));
+            var title = string.Format(EditorFeaturesResources._0_implemented_members,
+                FindUsagesHelpers.GetDisplayName(symbol));
 
-        using var _ = ArrayBuilder<DefinitionItem>.GetInstance(out var builder);
-        foreach (var impl in interfaceImpls)
+            using var _ = ArrayBuilder<DefinitionItem>.GetInstance(out var builder);
+            foreach (var impl in interfaceImpls)
+            {
+                builder.AddRange(await GoToDefinitionFeatureHelpers.GetDefinitionsAsync(
+                    impl, solution, thirdPartyNavigationAllowed: false, cancellationToken).ConfigureAwait(false));
+            }
+
+            var definitions = builder.ToImmutable();
+
+            return await _streamingPresenter.GetStreamingLocationAsync(
+                _threadingContext, solution.Workspace, title, definitions, cancellationToken).ConfigureAwait(false);
+        }
+
+        async ValueTask<INavigableLocation?> TryGetInterceptedLocationAsync()
         {
-            builder.AddRange(await GoToDefinitionFeatureHelpers.GetDefinitionsAsync(
-                impl, solution, thirdPartyNavigationAllowed: false, cancellationToken).ConfigureAwait(false));
+            if (symbol is not IMethodSymbol method)
+                return null;
+
+            // Find attributes of the form: [InterceptsLocationAttribute(version: 1, data: "...")];
+
+            var attributes = method.GetAttributes();
+            var interceptsLocationDatas = InterceptsLocationUtilities.GetInterceptsLocationData(attributes);
+            if (interceptsLocationDatas.Length == 0)
+                return null;
+
+            var lazyContentHashToDocumentMap = s_projectToLazyContentHashMap.GetValue(
```

Comment:
>s_projectToLazyContentHashMap [](http://example.com/codeflow?start=47&length=29)

Does this code commonly end up getting called repeatedly for the same project?

Reply:
I expect this to as i implement out more of the functionality for interceptors.  So I think the cache is worthwhile.